### DISCUSSION TITLE
Outer and Inner iterations

### DISCUSSION THREAD

**Discussion by simone-contorno** [Simple-Robotics/proxsuite#222] [Q&A] [answered]
Hello,
I'm using ProxQP to develop a Nonlinear Model Predictive Control using Sequential Quadratic Programming.
I would like to understand the difference between the outer and inner iterations set respectively by the parameters max_iter and max_iter_in. 
Thank you.

**Comment by jcarpent** [comment]
Thanks @simone-contorno for using ProxSuite.
@Bambade Could you provide a short answer?

**Comment by Bambade** [comment, marked as answer]
Hi @simone-contorno,

Thanks also for using ProxSuite!

ProxQP is based on a proximal point algorithm (PPA). In this context, **max_iter** corresponds to the maximum number of iterations allocated for running this PPA. As the QPs considered involve linear inequalities, the evaluation of the proximal operator requires running an inner loop (ProxQP runs precisely a Newton semi-smooth method with exact line-search for minimizing a strongly convex sub-problem). **max_iter_in** sets up the maximum number of inner loops for solving each proximal sub-problems.  

You can find more details in [our paper](https://www.roboticsproceedings.org/rss18/p040.pdf) if needs be. Feel free to ask other questions if you want

**Reply by simone-contorno** [reply]
You clearly explained the reason of these two loops, thank you!
